docs/theme/js/switcher.js

```diff
@@ -87,10 +87,36 @@
       return ''
     }
   
+    function nthIndex(str, pat, n){
+      var L= str.length, i= -1;
+      while(n-- && i++<L){
+          i= str.indexOf(pat, i);
+          if (i < 0) break;
+      }
+      return i;
+    }
+  
+    function count(str, pat){
+      var L= str.length, i= -1;
+      var n = 0;
+      while(i++<L){
+          i= str.indexOf(pat, i);
+          if (i < 0) break;
+          n++;
+      }
+      return n;
+    }
+
     function versions_root_in_url(url) {
-      var version = version_segment_in_url(url)
-      if(version == '') return get_host()
-      else return url.substring(0, url.indexOf(version))
+      var version = version_segment_in_url(url);
+      if(version == '') {
+        var depth_from_root = count(DOCUMENTATION_OPTIONS.URL_ROOT, "../");
```

Comment:
What exactly results in this being set differently for the daml on sql docs? I couldn’t find anything that could result in sphinx adding a different path here.

Reply:
This change is about being able to version the two docs independently. The old script first looked for a version string in the path so if you were at `https://example.com/foo/1.0.0/bar/baz.html`, it knew to look for `https://example.com/foo/versions.json` and replace the `1.0.0` with the selected version. As a fallback it looked for `https://example.com/versions.json`

The trouble now is that now you may be at `https://docs.daml.com/daml-driver-for-postgres/foo/bar.html` and need to figure out that your "root" is actually `https://docs.daml.com/daml-driver-for-postgres/`. Our theme already sets the `DOCUMENTATION_OPTIONS.URL_ROOT` property which is the relative path to the root of the sphinx tree. Afaict it's always `./` or `../` repeated. I can't use `https://docs.daml.com/daml-driver-for-postgres/foo/../` as the root, as that leads to bugs when trying to match on `root/version`. Ie I end up trying to match `https://docs.daml.com/daml-driver-for-postgres/foo/../` with `https://docs.daml.com/daml-driver-for-postgres/` which fails so I decided to just count depth and "navigate up".

Reply:
I get the change, what I don’t get is how `URL_ROOT` is set. The readme seems to be copied to `source/index.rst` which seems like it should be `./` not `../`.

Reply:
Oh, that's set through Sphinx's template engine: https://github.com/digital-asset/daml/pull/8188/files#diff-d4ebd498d60518b23a517eb9e53ca719f9536768a6d9eaa6fda1b314bf986807L48-L58

It's always the relative path to `/` from Sphinx's perspective, which is the `source` folder.

Reply:
And indeed, it's set to `./` in `index.html`.